tests/Jobs/RockTheVote/ImportRockTheVoteRecordTest.php

```diff
@@ -533,211 +533,284 @@ public function testSmsSubscriptionPayloadIsEmptyIfUserAlreadyHasMobile()
     }
 
     /**
-     * ---------------------------
-     * parseSmsStatusChangeForUser
-     * ---------------------------
-     */
-
-    /**
-     * Return mock user and row for a parseSmsStatusChange test.
+     * Return mock user and row that do not have voter registration changes for a
+     * SmsSubscriptionUpdate test.
      *
      * @param string $currentSmsStatus
      * @param bool @rtvSmsOptIn
      * @return obj
      */
-    public function getMocksForParseSmsStatusChangeTest($currentSmsStatus, bool $rtvSmsOptIn)
+    public function getMocksForUpdateUserSmsTest($currentSmsStatus, bool $rtvSmsOptIn)
     {
         $user = new NorthstarUser([
             'id' => $this->faker->northstar_id,
             'mobile' => $this->faker->phoneNumber,
             'sms_status' => $currentSmsStatus,
+            'sms_subscription_topics' => in_array($currentSmsStatus, [SmsStatus::$active, SmsStatus::$less, SmsStatus::$pending]) ? ['general', 'voting'] : [],
+            'voter_registration_status' => 'registration_complete',
         ]);
+
         $row = $this->faker->rockTheVoteReportRow([
             RockTheVoteRecord::$mobileFieldName => $this->faker->phoneNumber,
             RockTheVoteRecord::$smsOptInFieldName => $rtvSmsOptIn ? 'Yes' : 'No',
+            'Status' => 'Complete',
+            'Finish with State' => 'Yes',
         ]);
 
         return (object) ['user' => $user, 'row' => $row];
     }
 
     /**
-     * Tests SMS status is updated to active if user with null value opts-in.
-     *
      * @return void
      */
-    public function testParseSmsStatusChangeIfUserHasNullSmsStatusAndOptsIn()
+    public function testEnabledUpdateUserSmsWhenUserHasNullSmsStatusAndOptsIn()
     {
-        $mocks = $this->getMocksForParseSmsStatusChangeTest(null, true);
+        $this->enableUpdateUserSmsFeature(true);
+
+        $mocks = $this->getMocksForUpdateUserSmsTest(null, true);
```

Comment:
At first glance, I wasn't sure what `null, true` did here. I wonder if splitting this into two methods might be a bit more readable... maybe something like this? (It'd be even nicer if we could use model factories here, but I think that's probably a no-go since neither of these are actual Eloquent models...)

```php
$user = $this->getNorthstarUserMock(['sms_status' => null]);
$row = $this->getRockTheVoteRowMock($user, true);
```

Reply:
Definitely felt like I was losing readability by using one function to create both mocks, but it was also so tedious to write these tests, that having this function felt like a real timesaver. I'd love to split these out into unit and integration a little more some time, i could see creating a child class that could handle creating the mock user/rows for these subscription updates on setup to potentially DRY.